### PR TITLE
Add Flatpak

### PR DESCRIPTION
**Pull request by CharlieQLe**
This is an in-progress PR for a Flatpak build of ALVR for the Steam Flatpak. At the moment, it doesn't actually succeed because it can't find several dependencies, such as x264 and Cuda. It also targets the stable release at the moment, but that can be changed if needed.

Right now, the Flatpak pulls dependencies over the network during the build process. Ideally, the dependencies should be supplied within the manifest as to allow an offline build if needed. However, the Flatpak Cargo Generator script for generating the source manifests for all of the Cargo modules does not seem to work with Cargo modules fetched from Git repositories. If the script is updated to support this usecase, then the manifest should be updated to include the dependencies.

### CODE REVIEW

**Comment by CharlieQLe** [comment]
I've added x264, but the linker cites a ton of undefined references and fails to compile `alvr_server`.

File: packaging/flatpak/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+1. Install the Flatpak dependencies 
+
+```
+flatpak install flathub org.flatpak.Builder org.freedesktop.Sdk//22.08
+```
+
+2. Clone and enter this repository
+
+```
+git clone https://github.com/CharlieQLe/alvr-flatpak.git
```
Comment (Vixea) [inline packaging/flatpak/README.md:18]: Don't forget to edit this

**Comment by Vixea** [comment]
this needs to be moved into xtask and I think it would be fine to just leave lto off

**Review by zarik5 — CHANGES_REQUESTED** [review]
Documentation regarding the build of flatpak should go in "Build from source" page in the wiki.

**Comment by CharlieQLe** [comment]
> Documentation regarding the build of flatpak should go in "Build from source" page in the wiki.

I added a dedicated Flatpak wiki page. I feel that while building it could go into "Build from source", it would be better to have all the Flatpak information centralizing in one location where its shortcomings are listed, just so it isn't scattered across the wiki.

File: wiki/Flatpak.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+flatpak run --command=alvr_dashboard com.valvesoftware.Steam
+```
+
+A desktop file named `com.valvesoftware.Steam.Utility.alvr.desktop` is supplied within the `alvr/xtask/flatpak` directory. Move this to where other desktop files are located on your system in order to run the dashboard without the terminal.
+
+### Other Applications
+
+The support for other applications that are not launched via Steam is non-existent due to the Flatpak sandbox.
+
+Various SteamVR utilities such as [WlxOverlay](https://github.com/galister/WlxOverlay) and [OpenVR-AdvancedSettings](https://github.com/OpenVR-Advanced-Settings/OpenVR-AdvancedSettings) cannot run within the Flatpak sandbox due to their usage of AppImage. However, unpacking the supplied AppImage or building the utilities from source and running their binaries from within the sandbox similiarly to `alvr_dashboard` could work, but there is no guarantee that they will work properly.
```
Comment (Meister1593) [inline wiki/Flatpak.md:83]: Technically, i ran unpacked wlxoverlay into available dir from inside steam runtime (sh) and it ran almost flawlessly

It's not the same as installing to flatpak obviously, but as a workaround for now works

**Comment by Vixea** [comment]
I'm going to merge in it's current state within the next 24 hours, though for the desktop file we couldn't you just cp the file to correct location or use install?

**Comment by CharlieQLe** [comment]
> I'm going to merge in it's current state within the next 24 hours, though for the desktop file we couldn't you just cp the file to correct location or use install?

A script could be provided that copies the file and then launches the ALVR Dashboard, but it would have to run via the terminal once. The desktop file can't be copied to the host during the install as far as I know, and that just seems to be a limitation of Flatpak.

**Comment by Vixea** [comment]
it seems like you can do it using build-commands aka
```
      "build-commands": [
        "install -Dm644 flatpak/io.github.TTimo.GtkRadiant.desktop /app/share/applications/io.github.TTimo.GtkRadiant.desktop",
      ],
      ```
      https://github.com/flathub/flathub/pull/3682/files
      as in example, can you not do this?

**Comment by CharlieQLe** [comment]
That's not possible since `/app/share/applications` is read-only, and Flatpak creates all the symlinks to the desktop files when the application is installed, not an extension.

**Comment by Vixea** [comment]
Ok then I guess it will stay as is

**Review by zarik5 — APPROVED** [review]

**Comment by Vixea** [comment]
Thanks for your work, please take a break and enjoy you have done something that many alvr contributors have thought was impossible, ;D

**Pull request merged**